File: WMS-Service/src/main/java/net/ugurkartal/wmsservice/models/Product.java
package net.ugurkartal.wmsservice.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.With;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Document(collection = "products")
@Data
@AllArgsConstructor
@NoArgsConstructor
@With
public class Product {
    @Id
    private String id;
    private String name;
    private String description;
    private double salePrice;
    private int stock;
    private int criticalStock;
    private String imageUrl;
    @DBRef
    private Category category;
    private LocalDateTime created_at;
    private LocalDateTime updated_at;
    private boolean isActive;
}

File: wms-client/src/types/Product.ts
export type Product = {
    id: string,
    name: string,
    description: string,
    salePrice: number,
    stock: number,
    criticalStock: number,
    imageUrl: string,
    categoryName: string,
    createdAt: Date,
    updatedAt: Date,
    active: boolean
}